controllers/usersController.js

```diff
@@ -278,37 +290,158 @@ UsersController.resetPassword = function(req, res){
-                    res.status(404).send({
-                        operation: 'get',
-                        status: 'failed',
-                        message: 'User not found'
-                    });
+                    reject({code: 404, message: 'User not found in database'});
                 }
                 
+            }).catch(function(err){
+                reject({code: 500, message: "User could not be found due to: " + err});
             });
-
+    
         }).catch(function(err){
-            res.status(404).send({
-                operation: 'get',
-                status: 'failure',
-                message: 'Token was invalid'
-            });
+            reject({code: 400, message: 'Token was invalid; Message: ' + err})
         });
-    }
+
+    });
+
+    return promise;
+}
+
+/**
+ * Saves the application data to the user's profile
+ */
+UsersController.saveApplication = function(req, res){
+
+    editApplication(req.get("authorization"), req.body).then(function(result){
+        res.sendStatus(result.code);
+    }).catch(function(err){
+        res.status(err.code).send({
+            message: err.message
+        });
+    });
+
+}
+
+/**
+ * Saves the user's application, then handles additional logic which occurs after submitting an application
+ * e.g. sending a confirmation
+ */
+UsersController.submitApplication = function(req, res){
+
+    editApplication(req.get("authorization"), req.body).then(function(result){
+        return Database.get(COLLECTION_NAME, result.uid);
+    }).then(function(dbRes){
+        return Mail.addTag(MAILING_LIST, dbRes.email, "applied");
+    }).then(function(){
+        res.sendStatus(200);
+    }).catch(function(err){
+        let errCode = err.code || 500; // Use response code if it's passed down from editApplication, otherwise return a generic 500
+        res.status(errCode).send(err);
+    });
+
+}
+
+
+UsersController.getApplication = function(req, res){
+
+    let auth_header = req.get("authorization"); 
+    getUidFromHeader(auth_header).then(function(uid){
+
+        Database.get(COLLECTION_NAME, uid).then(function(databaseResult){
+        
+            // Check that the user has an application
+            // They should always have one, except for testing accounts
+            if(databaseResult && databaseResult.application){
+                res.status(200).send({
+                    operation: 'getApplication',
+                    status: 'success',
+                    data: databaseResult.application
+                });
+            } else {
+                res.status(404).send({
+                    operation: 'get',
+                    status: 'failed',
+                    message: 'User not found'
+                });
+            }
+            
+        });
+
+    }).catch(function(err){
+        res.status(404).send({
+            operation: 'get',
+            status: 'failure',
+            message: 'Token was invalid; Message: ' + err
+        });
```

Comment:
One line 314, you have a similar error, but you don't return the error to the client. I like it better that way.

Ideally anytime you have an error we would like for it to be logged, but we can change that later in a refactor.